File: src/Entity/Cour.php
<?php

namespace App\Entity;

use App\Repository\CourRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: CourRepository::class)]
class Cour
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le code du cours est requis.")]
    #[Assert\Length(min: 3, max: 255, minMessage: "Le code doit contenir au moins 3 caractères.")]
    private ?string $code_cours = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank(message: "Le titre est requis.")]
    #[Assert\Length(min: 3, max: 255, minMessage: "Le titre doit contenir au moins 3 caractères.")]
    private ?string $Titre = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank(message: "Le niveau est requis.")]
    #[Assert\Choice(choices: ["Débutant", "Intermédiaire", "Avancé"], message: "Le niveau doit être Débutant, Intermédiaire ou Avancé.")]
    private ?string $Niveau = null;

    #[ORM\Column(type: Types::TEXT)]
    #[Assert\NotBlank(message: "La catégorie est requise.")]
    private ?string $Categorie = null;

    #[ORM\ManyToOne(targetEntity: Module::class, inversedBy: "cours")]
    #[ORM\JoinColumn(nullable: false)]
    private ?Module $module = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCodeCours(): ?string
    {
        return $this->code_cours;
    }

    public function setCodeCours(string $code_cours): static
    {
        $this->code_cours = $code_cours;
        return $this;
    }

    public function getTitre(): ?string
    {
        return $this->Titre;
    }

    public function setTitre(string $Titre): static
    {
        $this->Titre = $Titre;
        return $this;
    }

    public function getNiveau(): ?string
    {
        return $this->Niveau;
    }

    public function setNiveau(string $Niveau): static
    {
        $this->Niveau = $Niveau;
        return $this;
    }

    public function getCategorie(): ?string
    {
        return $this->Categorie;
    }

    public function setCategorie(string $Categorie): static
    {
        $this->Categorie = $Categorie;
        return $this;
    }

    public function getModule(): ?Module
    {
        return $this->module;
    }

    public function setModule(?Module $module): static
    {
        $this->module = $module;
        return $this;
    }
}

File: Edu9raya-java/src/main/java/entities/Cour.java
package entities;

import java.sql.Timestamp;

public class Cour {

    private int id;
    private String codeCours;
    private String titre;
    private String niveau;
    private String categorie;
    private Module module; // Reference to Module (foreign key module_id)
    private String description;
    private String pdfPath;
    private Timestamp updatedAt;

    // Constructors
    public Cour() {
    }

    public Cour(String codeCours, String titre, String niveau, String categorie, Module module, String description, String pdfPath) {
        this.codeCours = codeCours;
        this.titre = titre;
        this.niveau = niveau;
        this.categorie = categorie;
        this.module = module;
        this.description = description;
        this.pdfPath = pdfPath;
    }

    public Cour(int id, String codeCours, String titre, String niveau, String categorie, Module module, String description, String pdfPath, Timestamp updatedAt) {
        this.id = id;
        this.codeCours = codeCours;
        this.titre = titre;
        this.niveau = niveau;
        this.categorie = categorie;
        this.module = module;
        this.description = description;
        this.pdfPath = pdfPath;
        this.updatedAt = updatedAt;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCodeCours() {
        return codeCours;
    }

    public void setCodeCours(String codeCours) {
        this.codeCours = codeCours;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getNiveau() {
        return niveau;
    }

    public void setNiveau(String niveau) {
        this.niveau = niveau;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public Module getModule() {
        return module;
    }

    public void setModule(Module module) {
        this.module = module;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPdfPath() {
        return pdfPath;
    }

    public void setPdfPath(String pdfPath) {
        this.pdfPath = pdfPath;
    }

    public Timestamp getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Timestamp updatedAt) {
        this.updatedAt = updatedAt;
    }

    @Override
    public String toString() {
        return "Cour{" +
                "id=" + id +
                ", codeCours='" + codeCours + '\'' +
                ", titre='" + titre + '\'' +
                ", niveau='" + niveau + '\'' +
                ", categorie='" + categorie + '\'' +
                ", moduleId=" + (module != null ? module.getId() : null) +
                ", description='" + description + '\'' +
                ", pdfPath='" + pdfPath + '\'' +
                ", updatedAt=" + updatedAt +
                '}';
    }
} 